### PR TITLE
[Merged by Bors] - Add multi draw indirect draw calls

### LINKED ISSUE #6216: Expose more `wgpu::RenderPass` draw APIs to `RenderCommand`s
## What problem does this solve or what need does it fill?

Users should be able to use the full extent of the `wgpu` draw APIs. While users have access to the `wgpu::RenderPass` via the `RenderContext` provided to `Node::run`, the `RenderCommand` API only has access to a `TrackedRenderPass`, which has a limited subset of draw APIs exposed; in particular, all of the `multi_draw...` variants are missing.

## What solution would you like?

Either add more wrapper methods to `TrackedRenderPass` or find some other abstraction that makes it feasible to work directly with the `wgpu::RenderPass` from the `RenderCommand` trait.

## What alternative(s) have you considered?

Implementing my own `Node` instead of using `RenderCommand`. This would be a major change.

### PR DESCRIPTION
**Pull request by Neo-Zhixing**
# Objective

- Allows bevy users to dispatch `multi_draw_indirect`, `multi_draw_indexed_indirect`,  `multi_draw_indirect_count`, `multi_draw_indexed_indirect_count` draw calls.
- Fixes #6216

## Solution

- Added the corresponding wrapper methods to `TrackedRenderPass`

---

## Changelog

> Added  `multi_draw_*` draw calls to `TrackedRenderPass`

### CODE REVIEW

**Review by IceSentry — APPROVED** [review]

File: crates/bevy_render/src/render_phase/draw_state.rs [hunk 1]
```diff
@@ -283,6 +283,166 @@ impl<'a> TrackedRenderPass<'a> {
...
+            "multi draw indirect: {:?} {}, {}x",
+            indirect_buffer,
+            indirect_offset,
+            count
+        );
+        self.pass
+            .multi_draw_indirect(indirect_buffer, indirect_offset, count);
+    }
+
+    /// Disptaches multiple draw calls from the active vertex buffer(s) based on the contents of the `indirect_buffer`.
```
Comment (vacuus) [inline crates/bevy_render/src/render_phase/draw_state.rs:319]: Dispatches

File: crates/bevy_render/src/render_phase/draw_state.rs [hunk 2]
```diff
@@ -283,6 +283,166 @@ impl<'a> TrackedRenderPass<'a> {
...
+    /// The count buffer is read to determine how many draws to issue.
+    ///
+    /// The indirect buffer must be long enough to account for `max_count` draws, however only `count` will
+    /// draws will be read. If `count` is greater than `max_count`, `max_count` will be used.
```
Comment (vacuus) [inline crates/bevy_render/src/render_phase/draw_state.rs:323]: What does `count` refer to? Draws are issued; "draws will be read" is inconsistent. I want to fold the second sentence into the first; maybe "`count` draws, capped by `max_count`, will be issued".

File: crates/bevy_render/src/render_phase/draw_state.rs (hunk 1 above)
Comment (Neo-Zhixing) [reply crates/bevy_render/src/render_phase/draw_state.rs]: good catch

**Pull request closed without merging**